src/shared/components/oncoprint/DeltaUtils.ts

```diff
@@ -405,10 +414,30 @@ function transitionTracks(
             }
         }
     }
-
+    
+    // collect max trackId (last assigned one) for each treatment profile
+    // the last assigned track id is used to reference the current layout of the track group
+    // Note: the resolution of `trackIds for ruleset sharing` is different from
+    // the section above because different formatting is applied to each treatment profile (molecularProfileId)
+    trackIdForRuleSetSharing.treatment = _.chain(prevProps.heatmapTracks)
+                                    .filter((s:IHeatmapTrackSpec) => s.molecularAlterationType === AlterationTypeConstants.TREATMENT_RESPONSE)
+                                    .groupBy((track:IHeatmapTrackSpec) => track.molecularProfileId)
+                                    .mapValues( (o:IHeatmapTrackSpec[]) => _(o).map('key').map((key:string) => trackSpecKeyToTrackId[key]).max() )
```

Comment:
we typically dont like to use the syntax of `.map("key")`, because I don't think typescript catches it. We use `.map(d=>d.key)`. Also on line 431

Reply:
In my dev setup it works as specified on the lodash docs. You probably are aware that this `map()` function is the lodash map and not the standard JS version? Should I go over all my changes and revert back all similar cases of `map('key')`?

Reply:
![image](https://user-images.githubusercontent.com/636232/55015942-2c7bb580-4fc4-11e9-9d6d-edb2539b5c3c.png)

![image](https://user-images.githubusercontent.com/636232/55015967-38677780-4fc4-11e9-8e40-8f0908544202.png)

Reply:
the functionality works, but typescript wont catch misspelled keys

Reply:
I have changed all occurences of the .map('key') pattern.